src/main/java/com/therockquarry/aes31/adl/TracklistSection.java

```diff
@@ -118,7 +121,7 @@ public void addTrackAtIndex (int trackNumber, TrackType track) throws InvalidDat
 			throw new InvalidDataException("Track \"" + track + "\" already exists in the Track List at index \"" + this.getTrackNumberForEntry(track) + "\".");
 		}
 		
-		_tracks.put(new Integer(trackNumber), track);
+		_tracks.put(key, track);
```

Comment:
It would seem that instead of `key`, this should be `trackNumber`. i.e.:
```
_tracks.put(trackNumber, track);
```

Reply:
I thought I got them all (there were LOTS like this). Time to kill this last one.

Reply:
There's anachronistic code here from an older version of Java where an int needed to be cast to an Integer. I wanted to do the least amount of changes as possible so `Integer key` was already established so continued using that variable.